hudi-common/src/main/java/org/apache/hudi/common/util/ClusteringUtils.java

```diff
@@ -124,7 +124,7 @@
         // get all filegroups in the plan
         getFileGroupEntriesInClusteringPlan(clusteringPlan.getLeft(), clusteringPlan.getRight()));
 
-    Map<HoodieFileGroupId, HoodieInstant> resultMap = resultStream.collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    Map<HoodieFileGroupId, HoodieInstant> resultMap = resultStream.collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (i1, i2) -> i1));
```

Comment:
so if I understand correctly, this will fix the duplicate key exception, but the repetitive scheduling is not yet fixed in this patch right ?

Reply:
yeah that's right. i thought about repetitive scheduling but such race scenarios are hard to avoid.. one way is as you suggested in the ticket i.e. to add some default `--min-sync-interval` but i felt that might give users the impression that job itself is taking longer than usual.